infrastructure/eks.tf

```diff
@@ -0,0 +1,31 @@
+module "eks" {
+  source          = "terraform-aws-modules/eks/aws"
+  version = "20.14.0"
+
+  cluster_name    = "Ognjen-cluster"
+  cluster_version = "1.29"
+  subnet_ids = concat(module.vpc.private_subnets, module.vpc.public_subnets)
```

Comment:
subnet_ids is where nodes will be provisioned, check documentation, it should not contain public subnets.

https://registry.terraform.io/modules/terraform-aws-modules/eks/aws/latest#input_subnet_ids